File: LC_Gen_Codes/2900_setting.py
class Solution:
    def getWordsInLongestSubsequence(self, n: int, words: List[str], groups: List[int]) -> List[str]:
        longest_subsequence = []
        for i in range(n):
            subsequence = [i]
            for j in range(i+1, n):
                if groups[subsequence[-1]] != groups[j]:
                    subsequence.append(j)
            if len(subsequence) > len(longest_subsequence):
                longest_subsequence = subsequence
        return [words[i] for i in longest_subsequence]

File: LC_Gen_Codes_Java/2900_setting.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> getWordsInLongestSubsequence(int n, String[] words, int[] groups) {
        List<String> longestSubsequence = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            List<Integer> subsequence = new ArrayList<>();
            subsequence.add(i);
            for (int j = i + 1; j < n; j++) {
                if (groups[subsequence.get(subsequence.size() - 1)] != groups[j]) {
                    subsequence.add(j);
                }
            }
            if (subsequence.size() > longestSubsequence.size()) {
                longestSubsequence = subsequence;
            }
        }
        List<String> result = new ArrayList<>();
        for (int index : longestSubsequence) {
            result.add(words[index]);
        }
        return result;
    }
}